src/sentry/api/bases/organization_events.py

```diff
@@ -312,42 +313,71 @@ def get_event_stats_data(
 
         with sentry_sdk.start_span(op="discover.endpoint", description="base.stats_serialization"):
             # When the request is for top_events, result can be a SnubaTSResult in the event that
             # there were no top events found. In this case, result contains a zerofilled series
             # that acts as a placeholder.
+            is_multiple_axis = len(query_columns) > 1
             if top_events > 0 and isinstance(result, dict):
                 results = {}
                 for key, event_result in result.items():
-                    if len(query_columns) > 1:
+                    if is_multiple_axis:
                         results[key] = self.serialize_multiple_axis(
-                            serializer, event_result, columns, query_columns, allow_partial_buckets
+                            serializer,
+                            event_result,
+                            columns,
+                            query_columns,
+                            allow_partial_buckets,
+                            zerofill_results=zerofill_results,
                         )
                     else:
                         # Need to get function alias if count is a field, but not the axis
                         results[key] = serializer.serialize(
                             event_result,
                             column=resolve_axis_column(query_columns[0]),
                             allow_partial_buckets=allow_partial_buckets,
+                            zerofill_results=zerofill_results,
                         )
-                return results
-            elif len(query_columns) > 1:
-                return self.serialize_multiple_axis(
-                    serializer, result, columns, query_columns, allow_partial_buckets
+                serialized_result = results
+            elif is_multiple_axis:
+                serialized_result = self.serialize_multiple_axis(
+                    serializer,
+                    result,
+                    columns,
+                    query_columns,
+                    allow_partial_buckets,
+                    zerofill_results=zerofill_results,
                 )
             else:
-                return serializer.serialize(
+                serialized_result = serializer.serialize(
                     result,
                     resolve_axis_column(query_columns[0]),
                     allow_partial_buckets=allow_partial_buckets,
+                    zerofill_results=zerofill_results,
                 )
+            if hasattr(result, "start") and hasattr(result, "end"):
+                if is_multiple_axis:
+                    for value in serialized_result.values():
+                        value["start"] = result.start
+                        value["end"] = result.end
+                else:
+                    serialized_result["start"] = result.start
+                    serialized_result["end"] = result.end
```

Comment:
Can this live inside of the `serialize` method of `SnubaTSResultSerializer`?

Reply:
👍  moved to the serialize method!